### DISCUSSION TITLE
Can we add TS types for the validator params?

### DISCUSSION THREAD

**Discussion by gerarts** [ing-bank/lion#1807] [Ideas]
Currently all validators accept `any` (through `type ValidatorParam = any`). Wouldn't it be better to strongly type all validator constructors?

We could even add a generic to Validator like:

```ts
export class Validator<P> extends EventTarget {
    constructor(param?: P, config?: import("../../types/validate").ValidatorConfig | undefined);
    __param: P;
    set param(arg: P);
    get param(): P;

    /* bunch of other stuff of course */
}
```

This would provide a better DX, and we might be able to remove some of the runtime checks (for example in string pattern), which are not implemented consistently in the current version (min- / max-string has no checks).

[MinLength Validator](https://github.com/ing-bank/lion/blob/d938e464b499bf48460829be0eeba596a41bd724/packages/ui/components/form-core/src/validate/validators/StringValidators.js#L52..L58) has no runtime checks

[Pattern Validator](https://github.com/ing-bank/lion/blob/d938e464b499bf48460829be0eeba596a41bd724/packages/ui/components/form-core/src/validate/validators/StringValidators.js#L129..L141) has runtime checks

**Comment by tlouisse** [comment]
Hey @gerarts, seems like a great improvement. Would you like to do a PR for this?
